src/main/flight/position.c

```diff
@@ -137,16 +142,44 @@ void calculateEstimatedAltitude(timeUs_t currentTimeUs)
     }
 #endif
 
-    if (ARMING_FLAG(ARMED) && !altitudeOffsetSet) {
+    if (ARMING_FLAG(ARMED) && !altitudeOffsetSetBaro) {
         baroAltOffset = baroAlt;
-        gpsAltOffset = gpsAlt;
-        altitudeOffsetSet = true;
-    } else if (!ARMING_FLAG(ARMED) && altitudeOffsetSet) {
-        altitudeOffsetSet = false;
+        altitudeOffsetSetBaro = true;
+    } else if (!ARMING_FLAG(ARMED) && altitudeOffsetSetBaro) {
+        altitudeOffsetSetBaro = false;
     }
+
     baroAlt -= baroAltOffset;
+
+    int goodGpsSats = 0;
+    int badGpsSats = -1;
+
+    if (haveBaroAlt) {
+        goodGpsSats = positionConfig()->altNumSatsGpsUse;
+        badGpsSats = positionConfig()->altNumSatsBaroFallback;
+        if (badGpsSats >= goodGpsSats) {
+            badGpsSats = goodGpsSats - 1;
+            positionConfigMutable()->altNumSatsBaroFallback = badGpsSats;
+        }
```

Comment:
With this change, if `goodGpsSats` is 4, and `badGpsSats` is 5, for example, will end with `badGpsSats` with a value of 3, that is outside the minimum for this variable.
I think is better to reset both to our default values and simplifies all.
I think remember other variables with similar problem have a "special" method that validates all the values, I will try to find the method in the code, to make it the same.

Reply:
I think is here:
https://github.com/betaflight/betaflight/blob/074172b4c94206e33a42d3aa5d8e4da2bd1a3e6d/src/main/config/config.c#L204